### PR TITLE
Documentation update

### PR DESCRIPTION
**Pull request by zveljkovic**
!!! Work in progress !!!
Updating documentation to be more noob friendly

### CODE REVIEW

File: docs/getting-started/installation.md [hunk 1]
```diff
@@ -109,14 +109,17 @@ Compose is currently published in two separate variants:
 
 === "Since v3.0.0-alpha01"
 
-    All Compose variants (Android, JVM/Desktop, Native/iOS, Native/macOS, JS/Canvas, Wasm) are published from the main branch, please add dependencies as usual (no version suffixes required).
+    All Compose variants (Android, JVM/Desktop, Native/iOS, Native/macOS, JS/Canvas, Wasm) are published from the main branch,
+    please add dependencies as usual (no version suffixes required).
```
Comment (arkivanov) [inline docs/getting-started/installation.md]: Unnecessary line wrap.

File: docs/getting-started/installation.md [hunk 2]
```diff
@@ -109,14 +109,17 @@ Compose is currently published in two separate variants:
...
 
 !!!warning
     Wasm target is not yet supported. Please follow [issue #74](https://github.com/arkivanov/Decompose/issues/74) for more information and updates.
 
 ## Extensions for Android views
 
-The `extensions-android` module provides extensions to connect Android views based UI to Decompose components. Please head to the corresponding [documentation page](../extensions/android.md) for more information.
+The `extensions-android` module provides extensions to connect
+Android views (older alternative to Compose UI) to Decompose components.
```
Comment (arkivanov) [inline docs/getting-started/installation.md]: Please write single line. You can enable Soft Wrap in the IDE for better readability.

File: mkdocs.yml [hunk 3]
```diff
@@ -111,8 +111,8 @@ markdown_extensions:
   - pymdownx.critic
   - pymdownx.details
   - pymdownx.emoji:
-      emoji_index: !!python/name:materialx.emoji.twemoji
-      emoji_generator: !!python/name:materialx.emoji.to_svg
+      emoji_index: !!python/name:material.extensions.emoji.twemoji
```
Comment (arkivanov) [inline mkdocs.yml:114]: Why is this change needed?

File: docs/getting-started/installation.md [hunk 4]
```diff
@@ -4,9 +4,21 @@ Decompose provides a number of modules, they are all published to Maven Central
 
 ## The main Decompose module
 
-The main functionality is provided by the `decompose` module. It contains some core features like [ComponentContext](../component/overview.md#componentcontext), [Child Stack](../navigation/stack/overview.md), etc.
+The main functionality is provided by the `decompose` module. It contains some core features like:
 
-### Gradle setup
+- [ComponentContext](../component/overview.md#componentcontext) needed to allow custom components
+to manage lifecycle, state saving,...
+- [Value](../component/overview.md/#value-and-mutablevalue-state-holders) allowing UI code to observe value changes
```
Comment (arkivanov) [inline docs/getting-started/installation.md]:
```suggestion
- [Value](../component/overview.md/#value-and-mutablevalue-state-holders) - Multiplatform (Swift-friendly) observable state holder (allows observing state changes in the UI).
```

File: docs/getting-started/installation.md [hunk 5]
```diff
@@ -4,9 +4,21 @@ Decompose provides a number of modules, they are all published to Maven Central
 
 ## The main Decompose module
 
-The main functionality is provided by the `decompose` module. It contains some core features like [ComponentContext](../component/overview.md#componentcontext), [Child Stack](../navigation/stack/overview.md), etc.
+The main functionality is provided by the `decompose` module. It contains some core features like:
 
-### Gradle setup
+- [ComponentContext](../component/overview.md#componentcontext) needed to allow custom components
+to manage lifecycle, state saving,...
```
Comment (arkivanov) [inline docs/getting-started/installation.md]:
```suggestion
- [ComponentContext](../component/overview.md#componentcontext) - provides API for components to manage lifecycle, state saving, etc.
```

File: docs/getting-started/installation.md [hunk 6]
```diff
@@ -4,9 +4,21 @@ Decompose provides a number of modules, they are all published to Maven Central
...
+- [Value](../component/overview.md/#value-and-mutablevalue-state-holders) allowing UI code to observe value changes
+- [Child Stack](../navigation/stack/overview.md) navigation model
```
Comment (arkivanov) [inline docs/getting-started/installation.md]:
```suggestion
- [Child Stack](../navigation/stack/overview.md) - stack navigation model.
```

File: docs/getting-started/installation.md [hunk 7]
```diff
@@ -4,9 +4,21 @@ Decompose provides a number of modules, they are all published to Maven Central
...
+- [defaultComponentContext](../getting-started/quick-start.md/#android-with-compose) function for generating root component context
+that is needed in UI code
```
Comment (arkivanov) [inline docs/getting-started/installation.md]:
```suggestion
- [defaultComponentContext](../getting-started/quick-start.md/#android-with-compose) - creates a default `ComponentContext` attached to an Activity or Fragment on Android.
```

File: docs/getting-started/installation.md [hunk 8]
```diff
@@ -4,9 +4,21 @@ Decompose provides a number of modules, they are all published to Maven Central
...
+- and many other important things
```
Comment (arkivanov) [inline docs/getting-started/installation.md]:
```suggestion
- And many other important APIs.
```

**Review by arkivanov — CHANGES_REQUESTED** [review]

File: docs/getting-started/installation.md (hunk 2 above)
Comment (zveljkovic) [reply docs/getting-started/installation.md]: Ok, will do. I did not use to Soft Wrap so will redo these.

File: mkdocs.yml (hunk 3 above)
Comment (zveljkovic) [reply mkdocs.yml:114]: When running latest mkdocs
mkdocs, version 1.5.3 from /usr/local/lib/python3.11/site-packages/mkdocs (Python 3.11)
there are deprecation warnings

INFO    -  DeprecationWarning: 'materialx.emoji.twemoji' is deprecated.
           Material emoji logic has been officially moved into mkdocs-material
           version 9.4. Please use Material's 'material.extensions.emoji.twemoji'
           instead of 'materialx.emoji.twemoji' in your 'mkdocs.yml' file.

File: docs/getting-started/installation.md (hunk 4 above)
Comment (zveljkovic) [reply docs/getting-started/installation.md]: Can we somehow make it more clear that this is the class that is used within both Android and iOS/Swift? 
For example 
"Multiplatform (Swift friendly) observable state holder (allows observing state changes in the UI)."

Comment (arkivanov) [reply docs/getting-started/installation.md]: Sure! Updated the suggestion.

File: docs/getting-started/installation.md (hunk 2 above)
Comment (zveljkovic) [reply docs/getting-started/installation.md]: This is changed

File: docs/getting-started/installation.md (hunk 1 above)
Comment (zveljkovic) [reply docs/getting-started/installation.md]: This is also done. Sorry for marking this earlier as resolved.

**Comment by zveljkovic** [comment]
I will continue working on the Android version of my app, and will update the docs in some other files.

File: docs/getting-started/installation.md [hunk 9]
```diff
@@ -38,55 +48,43 @@ Compose is currently published in two separate variants:
...
     === "Kotlin"
-    
+
         ``` kotlin
         implementation("com.arkivanov.decompose:extensions-compose-jetpack:<version>")
         // or
         implementation("com.arkivanov.decompose:extensions-compose-jetbrains:<version>")
         ```
+    Both modules are used to connect Compose with Decompose components. Please see the corresponding [documentation page](../extensions/compose.md).
+    Typically only one module should be selected, depending on the Compose variant being used.
```
Comment (arkivanov) [inline docs/getting-started/installation.md]:
```suggestion
    Both modules are used to connect Compose with Decompose components. Please see the corresponding [documentation page](../extensions/compose.md). Typically only one module should be selected, depending on the Compose variant being used.
```

File: docs/getting-started/installation.md [hunk 10]
```diff
@@ -4,9 +4,19 @@ Decompose provides a number of modules, they are all published to Maven Central
...
 
-### Gradle setup
+- [ComponentContext](../component/overview.md#componentcontext) - provides API for components to manage lifecycle, state saving, etc.
+- [Value](../component/overview.md/#value-and-mutablevalue-state-holders) - Multiplatform (Swift-friendly) observable state holder (allows observing state changes in the UI).
+- [Child Stack](../navigation/stack/overview.md) - stack navigation model.
+- [defaultComponentContext](../getting-started/quick-start.md/#android-with-compose) - creates a default `ComponentContext` attached to an Activity or Fragment on Android.
+- And many other important APIs.
+
+This module should be imported into build.gradle.kts for shared application module
+at ```kotlin.sourceSets.commonMain.dependencies```
```
Comment (arkivanov) [inline docs/getting-started/installation.md]:
```suggestion
This module should be imported into build.gradle.kts for shared application module at `kotlin.sourceSets.commonMain.dependencies`.
```

File: docs/getting-started/installation.md [hunk 11]
```diff
@@ -4,9 +4,19 @@ Decompose provides a number of modules, they are all published to Maven Central
...
+
+If you are having Android module (androidApp or composeApp if using KMP Wizard) this dependency also needs to be
+added into build.gradle.kts for Android module at ```android.dependencies``` to allow Android code access to above functionality
```
Comment (arkivanov) [inline docs/getting-started/installation.md]:
```suggestion
If you are having Android module (androidApp or composeApp if using KMP Wizard) this dependency also needs to be added into build.gradle.kts for Android module at ```android.dependencies``` to allow Android code access to above functionality
```

File: docs/getting-started/installation.md [hunk 12]
```diff
@@ -38,55 +48,43 @@ Compose is currently published in two separate variants:
...
-    
     === "Kotlin"
     
         ``` kotlin
         implementation("com.arkivanov.decompose:extensions-compose:<version>")
         ```
 
+    The module is used to connect Compose with Decompose components. Please see the corresponding [documentation page](../extensions/compose.md).
+
+
```
Comment (arkivanov) [inline docs/getting-started/installation.md]:
```suggestion

```

**Review by arkivanov — APPROVED** [review]

**Comment by arkivanov** [comment]
Thanks for updating the docs!

**Comment by zveljkovic** [comment]
Hi Arkadi, I see that you closed the PR, maybe we had misunderstanding regarding my last message. I meant that I have updated this installation docs and now will move to quick start while I try to implement basic functionality for the Android app. After that basic functionality is done I will try to make it all work for iOS and also update relevant docs. If it is ok, I will open another PR for the Android stuff when done, most probably later tonight.

**Comment by arkivanov** [comment]
Not closed but merged! 😀 The changes looked good, so why not to merge and do another PR later.
